src/SDL2.jl

```diff
@@ -78,16 +78,11 @@ module SDL2
         ev = Event(ntuple(i->UInt8(0),56))
         PollEvent(pointer_from_objref(ev)) == 0 && return nothing
 
-        t = UInt32(0)
-        for x in ev._Event[4:-1:1]
-            t = t << sizeof(x)*8
-            t |= x
-        end
+        ee = ev._Event
+        t = UInt32(ee[4]) << 24 | UInt32(ee[3]) << 16 | UInt32(ee[2]) << 8 | ee[1]
         evtype = Event(t)
         evtype == nothing && return nothing
 
-        evtype == KeyboardEvent && info(ev)
-
         unsafe_load( Ptr{evtype}(pointer_from_objref(ev)) )
```

Comment:
Are these `pointer_from_objref` correct, I never fully understood this Ref business, but it seem we should use a Ref somewhere ?

https://discourse.julialang.org/t/how-to-get-a-pointer-ptr-t-that-references-a-variable-of-type-t/9273

Reply:
Hmm, I'm not 100% sure either, but I believe that yes, that's an okay way to do what we want.

I _think_ it's essentially the `julia` equivalent of the following `C++`:
```c++
Foo f;
Bar&b = *reinterpret_cast<Bar*>(&f);
```

The information in that thread seems to be more about using `pointer_from_objref` as input to `ccall`, which it sounds like you shouldn't do. But in this case, I think it's doing what we want it to?

-----------------------

That said, this whole function does seem to be surprisingly slow on my machine. Even after removing the log line from the function in this PR, when I tried switching to this function, my game slowed down significantly. Before, I was simply reading the values I need out of the returned NTuple, in the same way we do to get `t` out of it. After switching to using this function to create the nice type-hierarchy `*Event` structs instead, the game got noticeably slower if I generated a lot of events (by moving the mouse around a lot, for instance).

I'm not sure which part is slow, perhaps its this cast business, or perhaps it's the type lookup in the dictionary? I should try to profile them I think but haven't gotten around to it...

Reply:
Do you have your game somewhere, I would be curious to have a look ?

Reply:
Hey, yeah, sorry! I finally uploaded it. I didn't have it up when i commented last cause i wasn't sure what license to put on it. :) it's just a simple little pong clone, but it's here!!!:
https://github.com/NHDaly/PaddleBattleJL

It was a lot of fun to make it. So much more fun than doing SDL in C++ ever was. Especially editing it in Juno through Atom was fantastic: being able to just `ctrl-c` then edit a function and reexecute it, and then just `shift-enter` on [`runSceneGameLoop`](https://github.com/NHDaly/PaddleBattleJL/blob/master/main.jl#L537) to restart playing with the new code was _so nice_. :) Thanks for your help in putting it together!

Reply:
Awesome, it's quit impressive and plays very smoothly.

Btw if you are into this kind of things, I initially made this package to try to build a more abstract way of making simple GUI apps in Julia, the idea is to support both the SDL2+Cairo and OpenGL backends, abstracting away the nasty details, but I haven't got very far with it:

https://github.com/jonathanBieler/GUIFrameworkPlayground.jl/blob/master/examples/example01.jl

Reply:
> Awesome, it's quit impressive and plays very smoothly.

:) Thanks! It was fun to make. the actual game only took like a day or two, but making it smooth and polished took weeks. :'(  #gamedevelopment

Reply:
> Btw if you are into this kind of things, I initially made this package to try to build a more abstract way of making simple GUI apps in Julia, the idea is to support both the SDL2+Cairo and OpenGL backends, abstracting away the nasty details, but I haven't got very far with it:
> 
> https://github.com/jonathanBieler/GUIFrameworkPlayground.jl/blob/master/examples/example01.jl

:D That's really awesome! It looks cool. Yeah that seems like a nicely abstracted interface.
Keep at it! I think writing your own GUI framework is superrrrr educational and exciting. I actually attempted the same thing a bunch of years ago, in C++, using SDL as well!:
https://github.com/NHDaly/SDL-GUI-Library

(I was using SDL v1.2 I think.) There ended up being _so many_ design decisions to make: how should Views be nested, how should a user register their callbacks, how do you handle user exceptions, etc. It was lots of fun. :)
Unfortunately I don't think I have the time to help with something like that now, but I'm always around if you wanna chat / bounce ideas off of me! :D

Reply:
:) Just wanted to share this writeup with you!:
https://discourse.julialang.org/t/building-and-releasing-a-fully-self-contained-application-written-in-julia/10418

I "finished" Paddle Battle! It's basically the same as a month ago, i just finally managed to sign it. 😅

Thanks again for your help! :)